src/view/components/microbit/MicrobitSimulator.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from "react";
+import { MicrobitImage } from "./MicrobitImage";
+
+const initialLedState = [
+    [0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0],
+];
+export class MicrobitSimulator extends React.Component<any, { leds: any }> {
+    constructor() {
+        super({});
+        this.state = { leds: initialLedState };
+    }
+    handleMessage = (event: any): void => {
+        const message = event.data;
+        switch (message.command) {
+            case "reset-state":
+                console.log("Reset the state");
+                this.setState({
+                    leds: initialLedState,
+                });
+                break;
+            case "set-state":
+                console.log("Setting the state");
+                this.setState({
+                    leds: message.state.leds,
+                });
+                break;
+        }
+    };
+    componentDidMount() {
+        window.addEventListener("message", this.handleMessage);

```

Comment:
Do you know if this could be triggered by something else unintentionally (maybe VS Code)? If yes, are we making sure this is a no-op (or whatever might be a proper response)?

Reply:
If vscode sends other kind of message? In my next PR I add some verification in this.handleMessage to only use logic if the device corresponds to the selected device (Ex: only read microbit messages if the message is tagged microbit